library/src/scala/tasty/reflect/CompilerInterface.scala

```diff
@@ -931,36 +931,18 @@ trait CompilerInterface {
 
   def matchTermRef(tpe: TypeOrBounds) given (ctx: Context): Option[TermRef]
 
-  def matchTermRef_unapply(tpe: TypeOrBounds) given (ctx: Context): Option[(Symbol, TypeOrBounds /* Type | NoPrefix */)]
+  def TermRef_apply(qual: TypeOrBounds, name: String) given (ctx: Context): TermRef
```

Comment:
Take a `Symbol` instead of Name, for safety?

Reply:
That may incur a bigger refactoring. I will try it out in a different PR